### DISCUSSION TITLE
Build with corporate proxy

### DISCUSSION THREAD

**Discussion by Xan2063** [strimzi/strimzi-kafka-operator#4808] [Q&A]
Hi everyone,

I am trying to configure the build for Kafka Connect behind a corporate proxy. I tried the following approaches, but none of the seem to work

* set the http_proxy and https_proxy env var for the build container template
* set the BUILD_ARGS env var for the build container env var
* Trying to set the additionalKaniko arguments to --build-args http_proxy=proxy:8080

The last one creates an error message that this is an illegal parameter, the first one fail on the curl in the build container.

Am I missing something to get the proxy in?

Thanks for your work!

**Comment by scholzj** [comment]
I do not think this was ever tested with a proxy. How do you build other containers behind proxy? Or how would you build the generated Dockerfile behind your proxy?

**Reply by Xan2063** [reply]
Usually we pass the proxy configuration to the docker process and set the proxy inside the Docker file or pass them as arguments to the build command.

**Reply by scholzj** [reply]
Ok, I created #4814 for this and will try to get to it. But not sure this will happen before 0.23 is released I'm afraid :-(.

**Reply by Xan2063** [reply]
Thank you, commented a bit more on the issue. I am now building the image myself and just reference it, so everything works for now. 

I tried creating a proxy image, setting this one as image to the connect cluster and then building on top of this. But the build container then keeps claiming he wants to download a null image from docker. So no luck on this one either.
